Sources/Sentry/Public/SentryDefines.h

```diff
@@ -1,5 +1,11 @@
 #import <Foundation/Foundation.h>
 
+// SentryDefines.h is a key header and will be checked early,
+// ensuring this error appears first during the compile process.
+#if APPLICATION_EXTENSION_API_ONLY_NO
+#    error Set APPLICATION_EXTENSION_API_ONLY to YES in the Sentry build settings.
```

Comment:
`m`: Please add a more detailed explanation of why this is required and maybe add a link to the troubleshooting guide. You can do this like that
```
#    error "Set APPLICATION_EXTENSION_API_ONLY to YES in the Sentry build settings. \
#      Some comment \
#      For more information, visit https://docs.sentry.io/platforms/apple/"
```

Reply:
Added in https://github.com/getsentry/sentry-cocoa/pull/4603/commits/3164dd6faee1522cae147f7e69794ab02f54e54e